kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/server/AdbServerImpl.kt

```diff
@@ -30,58 +30,36 @@ class AdbServerImpl(
             return AdbTerminal
         }
 
-    /**
-     * Executes shell commands blocking current thread.
-     * Please be aware! If any command that is in @param commands failed then AdbServerException will be thrown
-     *
-     * Required Permissions: INTERNET.
-     *
-     * @param commands commands to execute.
-     * @throws AdbServerException if a result status of any command in @param commands is Failed
-     * @return list of answers of commands' execution
-     */
     override fun performCmd(vararg commands: String): List<String> {
         return perform(commands) {
             adbTerminal.executeCmd(it)
         }
     }
 
-    /**
-     * Performs adb commands blocking current thread.
-     * Please be aware! If any command that is in @param commands failed then AdbServerException will be thrown
-     *
-     * Required Permissions: INTERNET.
-     *
-     * @param commands commands to execute.
-     * @throws AdbServerException if a result status of any command in @param commands is Failed
-     * @return list of answers of commands' execution
-     */
     override fun performAdb(vararg commands: String): List<String> {
         return perform(commands) {
             adbTerminal.executeAdb(it)
         }
     }
 
-    /**
-     * Performs shell commands blocking current thread.
-     * Please be aware! If any command that is in @param commands failed then AdbServerException will be thrown
-     *
-     * Required Permissions: INTERNET.
-     *
-     * @param commands commands to execute.
-     * @throws AdbServerException if a result status of any command in @param commands is Failed
-     * @return list of answers of commands' execution
-     */
     override fun performShell(vararg commands: String): List<String> {
         return perform(commands) {
             adbTerminal.executeAdb("shell $it")
         }
     }
 
-    /**
-     * Disconnect from AdbServer.
-     * The method is called by Kaspresso after each test.
-     */
+    override fun performCmd(command: String, arguments: List<String>): String {
+        return performComplex(command, arguments, adbTerminal::executeCmd)
```

Comment:
I guess it's better to use the single `perform` method. Don't see why we need `performComplex` method.